### PR TITLE
CEDS-2013 - skip country of routing for clearance request

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/controllers/declaration/DestinationCountryController.scala [hunk 1]
```diff
@@ -67,9 +67,9 @@ class DestinationCountryController @Inject()(
 
   private def redirectToNextPage(mode: Mode)(implicit request: JourneyRequest[AnyContent]): Result =
     request.declarationType match {
-      case DeclarationType.SUPPLEMENTARY =>
+      case DeclarationType.SUPPLEMENTARY | DeclarationType.CLEARANCE =>
         navigator.continueTo(controllers.declaration.routes.LocationController.displayPage(mode))
-      case DeclarationType.STANDARD | DeclarationType.SIMPLIFIED | DeclarationType.OCCASIONAL | DeclarationType.CLEARANCE =>
+      case DeclarationType.STANDARD | DeclarationType.SIMPLIFIED | DeclarationType.OCCASIONAL =>
```
Comment (ed0906) [inline app/controllers/declaration/DestinationCountryController.scala]: Do we need to skip the back end mapping for this too?

File: test/views/declaration/LocationViewSpec.scala [hunk 2]
```diff
@@ -401,4 +401,39 @@ class LocationViewSpec extends UnitViewSpec with ExportsTestData with Stubs with
...
+        "display 'Back' button that links to correct page" in {
+          val backButton = view.getElementById("back-link")
+
+          backButton must containText("site.back")
+          backButton.getElementById("back-link") must haveHref(redirect)
+        }
+      }
+
+    "for Standard declaration" should {
+      behave like viewWithCorrectBackButton(DeclarationType.STANDARD, controllers.declaration.routes.RoutingCountriesSummaryController.displayPage())
```
Comment (ed0906) [inline test/views/declaration/LocationViewSpec.scala]: `behave like` thats new

What's that do?

File: app/controllers/declaration/DestinationCountryController.scala (hunk 1 above)
Comment (SteveSugden) [reply app/controllers/declaration/DestinationCountryController.scala]: What do you mean @ed0906 ?

Comment (ed0906) [reply app/controllers/declaration/DestinationCountryController.scala]: The back end is set up to map clearance requests like  standards is it not?
If we are skipping pages, does that mean we can remove that part of the mapping from the back end for this journey type?

File: test/views/declaration/LocationViewSpec.scala (hunk 2 above)
Comment (SteveSugden) [reply test/views/declaration/LocationViewSpec.scala]: It's not that new - it's used in a couple of places.   Basically used to create parameterised blocks of code that can be re-used

File: app/controllers/declaration/DestinationCountryController.scala (hunk 1 above)
Comment (PatrykRudnicki) [reply app/controllers/declaration/DestinationCountryController.scala]: I think that those fields are optional in the model, so it's not a problem that backend model has everything

File: app/controllers/declaration/RoutingCountriesSummaryController.scala [hunk 3]
```diff
@@ -46,18 +46,13 @@ class RoutingCountriesSummaryController @Inject()(
...
-      case _ =>
-        val countryCodes = request.cacheModel.locations.routingCountries
-        val countries = findByCodes(countryCodes)
-
-        if (countries.nonEmpty) {
-          Ok(routingCountriesSummaryPage(mode, RoutingQuestionYesNo.form(), countries))
-        } else {
-          navigator.continueTo(controllers.declaration.routes.RoutingCountriesController.displayRoutingQuestion(mode))
-        }
+    val countryCodes = request.cacheModel.locations.routingCountries
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/RoutingCountriesSummaryController.scala]: What about case that you removed? This screen is not a part of the `SUPPLEMENTARY` journey

**Pull request merged**